Kickstarter-iOS/Features/PledgePaymentMethods/Controller/PledgePaymentMethodsViewController.swift

```diff
@@ -223,6 +222,98 @@ final class PledgePaymentMethodsViewController: UIViewController {
     }
   }
 
+  private func didConfirmAddedPaymentOption(
+    paymentOption: PaymentSheet.FlowController.PaymentOptionDisplayData,
+    forClientSecret clientSecret: String,
+    forPaymentSheetType paymentSheetType: PledgePaymentSheetType
+  ) {
+    // Stripe also defines this constant, but it's defined internally to the Stripe SDK :(
+    let linkIdentifier = "link"
+
+    // For regular payment types, continue with the display information they gave us
+    if paymentOption.paymentMethodType != linkIdentifier {
+      let paymentDisplayData = PaymentSheetPaymentOptionsDisplayData(
+        image: paymentOption.image,
+        label: paymentOption.label
+      )
+
+      self.viewModel.inputs.paymentSheetDidAdd(newCard: paymentDisplayData, clientSecret: clientSecret)
+      return
+    }
+
+    // For link payment types, fetch the underlying payment method so we can get at the card type and label
+
+    switch paymentSheetType {
+    case .setupIntent:
+      STPAPIClient.shared
+        .retrieveSetupIntent(
+          withClientSecret: clientSecret,
+          expand: ["payment_method"]
+        ) { [weak self] intent, _ in
```

Comment:
The callbacks for `retrieveSetupIntent` and for `retrievePaymentIntent` have different types, so they can't be shared.